prow/sidecar/run.go

```diff
@@ -234,8 +239,9 @@ func (o Options) doUpload(ctx context.Context, spec *downwardapi.JobSpec, passed
 		logrus.WithField("duration", time.Since(startTime).String()).Info("Finished uploading")
 	}()
 
-	for logName, reader := range logReaders {
-		uploadTargets[logName] = gcs.DataUpload(reader)
+	for logName, readerFunc := range logReadersFuncs {
+		readerFunc := readerFunc
```

Comment:
I don't think this is necessary. The type of `logReadersFuncs` is `map[string]gcs.ReaderFunc`, in which I believe that the `gcs.readerFunc` pointer be different on every cycle. Did you observe something different?

Reply:
Good catch, thanks! `readerFunc` passes through several functions call before being capture by a closure, that's avoid variable aliasing.